autoPyTorch/api/base_task.py

```diff
@@ -1079,73 +1135,223 @@ def refit(
+                for each finished configuration. This argument allows the user to skip
+                saving certain file type, for example the model, from being written to disk.
+            configuration: (Optional[Configuration])
                 configuration to fit the pipeline with. If None,
-                uses default
+                uses default.
 
         Returns:
             (BasePipeline): fitted pipeline
+            (RunInfo): Run information
+            (RunValue): Result of fitting the pipeline
+            (BaseDataset): Dataset created from the given tensors
         """
-        if self.dataset_name is None:
-            self.dataset_name = str(uuid.uuid1(clock_seq=os.getpid()))
 
-        if self._logger is None:
-            self._logger = self._get_logger(self.dataset_name)
+        resampling_strategy = resampling_strategy if resampling_strategy is not None else self.resampling_strategy
+        resampling_strategy_args = resampling_strategy_args if resampling_strategy_args is not None else \
+            self.resampling_strategy_args
+
+        dataset = self._create_dataset(X_train=X_train,
+                                       y_train=y_train,
+                                       X_test=X_test,
+                                       y_test=y_test,
+                                       resampling_strategy=resampling_strategy,
+                                       resampling_strategy_args=resampling_strategy_args,
+                                       dataset_name=dataset_name,
+                                       return_only=True)
 
         # get dataset properties
         dataset_requirements = get_dataset_requirements(
             info=self._get_required_dataset_properties(dataset))
         dataset_properties = dataset.get_dataset_properties(dataset_requirements)
         self._backend.save_datamanager(dataset)
 
+        self._backend._make_internals_directory()
+
+        if self._logger is None:
+            self._logger = self._get_logger(dataset.dataset_name)
+
         # build pipeline
-        pipeline = self.build_pipeline(dataset_properties)
-        if pipeline_config is not None:
-            pipeline.set_hyperparameters(pipeline_config)
+        if include_components is None:
+            include_components = self.include_components
+        if exclude_components is None:
+            exclude_components = self.exclude_components
+        if search_space_updates is None:
+            search_space_updates = self.search_space_updates
+
+        pipeline = self.build_pipeline(dataset_properties=dataset_properties,
+                                       include_components=include_components,
+                                       exclude_components=exclude_components,
+                                       search_space_updates=search_space_updates)
+        if configuration is None:
+            configuration = pipeline.get_hyperparameter_search_space().get_default_configuration()
```

Comment:
Do you think it makes sense to make this more flexible? Why the default and not a random. Maybe we make configuration a required argument? Shouldn't this be passed to the build_pipeline?

Reply:
So I think I'll remove this part, and make configuration a required argument. This will then be passed to the TAE and that will take care of the rest.